src/main/java/com/gotocompany/firehose/evaluator/GrpcResponseCELPayloadEvaluator.java

```diff
@@ -0,0 +1,66 @@
+package com.gotocompany.firehose.evaluator;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.Message;
+import lombok.extern.slf4j.Slf4j;
+import org.projectnessie.cel.checker.Decls;
+import org.projectnessie.cel.tools.Script;
+import org.projectnessie.cel.tools.ScriptCreateException;
+import org.projectnessie.cel.tools.ScriptException;
+import org.projectnessie.cel.tools.ScriptHost;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public class GrpcResponseCELPayloadEvaluator implements PayloadEvaluator<Message> {
+
+    private final String celExpression;
+    private Script script;
+    private Descriptors.Descriptor descriptor;
+
+    public GrpcResponseCELPayloadEvaluator(Descriptors.Descriptor descriptor, String celExpression) {
+        this.celExpression = celExpression;
+        this.descriptor = descriptor;
+        this.script = buildScript(descriptor);
+    }
+
+    @Override
+    public boolean evaluate(Message payload) {
+        try {
+            Map<String, Object> arguments = new HashMap<>();
+            arguments.put(payload.getDescriptorForType().getFullName(), payload);
+            return getScript(payload.getDescriptorForType()).execute(Boolean.class, arguments);
+        } catch (ScriptException e) {
+            throw new IllegalArgumentException(
+                    "Failed to evaluate payload with CEL Expression with reason: " + e.getMessage(), e);
+        }
+    }
+
+    private Script getScript(Descriptors.Descriptor payloadDescriptor) throws ScriptCreateException {
+        if (!payloadDescriptor.equals(this.descriptor)) {
+            synchronized (this) {
+                if (!payloadDescriptor.equals(this.descriptor)) {
+                    this.script = buildScript(payloadDescriptor);
+                    this.descriptor = payloadDescriptor;
+                }
+            }
+        }
+        return this.script;
+    }
+
+    private Script buildScript(Descriptors.Descriptor payloadDescriptor) {
+        try {
+            log.info("Building new CEL Script");
+            return ScriptHost.newBuilder()
```

Comment:
no need to refresh the script, since if the rule change we need to supply new CEL Expression

Reply:
if the descriptor is updated, if we use the old descriptor in this script, will it work?